src/firefly/java/edu/caltech/ipac/firefly/server/catquery/CatMasterTableQuery.java

```diff
@@ -35,110 +35,48 @@
-            File catOutFile = MASTER_CAT_FILE;
-
-            // if hydra, check for additional catalogs
-            DataGroup dgExtra = null;
-            String projectId = request.getParam("projectId");
-//            if (!StringUtils.isEmpty(projectId)) {
-//                ProjectTag pTag = DynConfigManager.getInstance().getCachedProject(projectId);
-//                List<CatalogTag> cList = pTag.getCatalogs();
-//                for (CatalogTag cTag : cList) {
-//                    String fullCatalogUrl = cTag.getHost() + cTag.getCatalogUrl();
-//                    File catInFile = new File(ServerContext.getPermWorkDir(), cTag.getOriginalFilename());
-//                    placeOriginal(fullCatalogUrl, catInFile);
-//
-//                    catOutFile = new File(ServerContext.getPermWorkDir(), cTag.getMasterCatFilename());
-//
-//                    String selectStr =
-//                            "select col " +
-//                                    colNames + " " +
-//                                    "from " + catInFile.getPath() + " " +
-//                                    "with complete_header";
-//                    DataGroupQueryStatement statement = DataGroupQueryStatement.parseStatement(selectStr);
-//                    DataGroup dg = statement.execute();
-//
-//                    if (dgExtra == null) {
-//                        dgExtra = dg;
-//
-//                    } else {
-//                        // concat dg to dgExtra
-//                        Iterator<DataObject> iter = dg.iterator();
-//                        while (iter.hasNext()) {
-//                            dgExtra.add(iter.next());
-//                        }
-//                    }
-//                }
-//            }
-
-            // get all other catalog data
-            placeOriginal(MASTER_LOC, ORIGINAL_FILE);
-
-            String selectStr =
-                    "select col " +
-                            colNames + " " +
-                            "from " + ORIGINAL_FILE.getPath() + " " +
-                            "with complete_header";
-            DataGroupQueryStatement statement = DataGroupQueryStatement.parseStatement(selectStr);
-            DataGroup dg = statement.execute();
 
-            DataGroup outputDG;
-            if (dgExtra == null) {
-                outputDG= dg;
+    public DataGroup fetchDataGroup(TableServerRequest req) throws DataAccessException {
 
-            } else {
-                // concat dg to dgExtra
-                Iterator<DataObject> iter = dg.iterator();
-                while (iter.hasNext()) {
-                    dgExtra.add(iter.next());
+        AtomicReference<DataGroup> dgRef = new AtomicReference<>();
+        AtomicReference<DataAccessException> exRef = new AtomicReference<>();
+        HttpServices.getData(new HttpServiceInput(MASTER_LOC), method -> {
```

Comment:
I think you should keep using URLDownload.  The way out stuff has more consistent logging. 
Eventually we might want to replace the insides of URLDownload with HttpServices.

Reply:
What I've done is creating a DataGroup from the response's OutputStream directly without writing to disk.
I need to be able to handle the response's OutputStream directly.
I don't think it's possible to do this with URLDownload as it is written.

Reply:
ok

Reply:
I still have a general logging concern.  I have recently been investigating problems and find URL download easy to track while using HttpService more challenging.

We should think about it in the future.